### ISSUE TITLE
MULTISET emulation using XML produces a null Result

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13080]
When emulating a `MULTISET()` operator using XML or JSON, the behaviour with respect to empty sets is different. The expectation is for `MULTISET()` to always produce a non-null value whereas `MULTISET_AGG()` should copy SQL's annoying `NULL`-on-empty-set-aggregation behaviour. For example, in PostgreSQL:

```sql
select
  array(select 1 where false),
  (select array_agg(1) where false)
)
```

This produces:

```
|array|array_agg|
|-----|---------|
|{}   |NULL     |
```

The JSON emulation of `MULTISET` has a `COALESCE()` operator:

```sql
select
  t_language.ID,
  t_language.CD,
  (
    select coalesce(
      json_agg(json_build_array(
        v0, ...
      )),
      json_build_array()
    )
    from (
      select
        t_book.ID as v0, ...
      from t_book
      where t_book.LANGUAGE_ID = t_language.ID
    ) as t
  )
from t_language
order by t_language.ID
```

While the SQL emulation doesn't require this `COALESCE` operation (see also https://github.com/jOOQ/jOOQ/issues/12259)

```sql
select
  t_language.ID,
  t_language.CD,
  (
    select xmlelement(
      name result,
      xmlagg(xmlelement(
        name record,
        xmlelement(name v0, v0), ...
      ))
    )
    from (
      select
        t_book.ID as v0, ...
      from t_book
      where t_book.LANGUAGE_ID = t_language.ID
    ) as t
  )
from t_language
order by t_language.ID
```

... it produces `<result/>`, which isn't deserialised correctly, so we get a `(Result) null` value nonetheless

**Comment by lukaseder** [comment]
Users may have started to rely on the emulation's behaviour, so this is an incompatible change.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0. No backport is planned, due to the possibly unexpected change of behaviour.

**Comment by lukaseder** [comment]
(Most people probably use the default `JSON` multiset emulation)
